metrics/api/src/main/java/io/helidon/metrics/api/RegistryFactoryManager.java

```diff
@@ -105,21 +109,36 @@ static RegistryFactory getInstance() {
         return INSTANCE.get();
     }
 
-    static synchronized RegistryFactory getInstance(MetricsSettings metricsSettings) {
-        RegistryFactoryManager.metricsSettings = metricsSettings;
-        RegistryFactory result = INSTANCE.get();
-        result.update(metricsSettings);
-        return result;
+    static RegistryFactory getInstance(MetricsSettings metricsSettings) {
+
+        return accessMetricsSettings(() -> {
+            RegistryFactoryManager.metricsSettings = metricsSettings;
+            RegistryFactory result = INSTANCE.get();
+            result.update(metricsSettings);
+            return result;
+        });
     }
 
-    static synchronized RegistryFactory getInstance(ComponentMetricsSettings componentMetricsSettings) {
-        return componentMetricsSettings.isEnabled()
+    static RegistryFactory getInstance(ComponentMetricsSettings componentMetricsSettings) {
+
+        return accessMetricsSettings(() -> componentMetricsSettings.isEnabled()
                 ? INSTANCE.get()
-                : NO_OP_INSTANCE;
+                : NO_OP_INSTANCE);
     }
 
     @Deprecated
     static RegistryFactory getInstance(Config config) {
         return getInstance(MetricsSettings.create(config));
     }
+
+    private static <T> T accessMetricsSettings(Supplier<T> operation) {
+        try {
+            SETTINGS_ACCESS.acquire();
+            T result = operation.get();
+            SETTINGS_ACCESS.release();
```

Comment:
This will not occur if `operation.get()` fails.

Reply:
Fixed